src/state/lists/hooks.ts

```diff
@@ -47,13 +48,16 @@ export function useAllLists(): AppState['lists']['byUrl'] {
 }
 
 function combineMaps(map1: TokenAddressMap, map2: TokenAddressMap): TokenAddressMap {
-  return {
-    [1]: { ...map1[1], ...map2[1] },
-    [4]: { ...map1[4], ...map2[4] },
-    [3]: { ...map1[3], ...map2[3] },
-    [42]: { ...map1[42], ...map2[42] },
-    [5]: { ...map1[5], ...map2[5] },
-  }
+  return Object.values(SupportedChainId).reduce(
```

Comment:
can we make the improvement of returning a map that only contains chains from map1 or map2 and add unit tests? also there's potential perf savings here of not using the spread operator when an object only exists on map1 or map2

Reply:
you might be able to simplify the logic with [_merge](https://lodash.com/docs/4.17.15#merge)